Riot/Modules/TabBar/MasterTabBarController.m

```diff
@@ -494,112 +497,122 @@ - (void)onMatrixSessionStateDidChange:(NSNotification *)notif
+    onboardingCoordinatorBridgePresenter.completion = ^{
+        MXStrongifyAndReturnIfNil(self);
+        [self.onboardingCoordinatorBridgePresenter dismissWithAnimated:YES completion:nil];
+        self.onboardingCoordinatorBridgePresenter = nil;
+        
+        self.isOnboardingInProgress = NO;
+        [self.masterTabBarDelegate masterTabBarControllerDidCompleteAuthentication:self];
+    };
     
-    // Keep ref on the authentification view controller while it is displayed
-    // ie until we get the notification about a new account
-    _authViewController = authenticationViewController;
-    isAuthViewControllerPreparing = NO;
+    [onboardingCoordinatorBridgePresenter presentFrom:self animated:NO];
     
-    // Listen to the end of the authentication flow
-    _authViewController.authVCDelegate = self;
+    self.onboardingCoordinatorBridgePresenter = onboardingCoordinatorBridgePresenter;
+    self.isOnboardingCoordinatorPreparing = NO;
     
-    authViewControllerObserver = [[NSNotificationCenter defaultCenter] addObserverForName:kMXKAccountManagerDidAddAccountNotification object:nil queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *notif) {
-        
-        _authViewController = nil;
+    self.addAccountObserver = [[NSNotificationCenter defaultCenter] addObserverForName:kMXKAccountManagerDidAddAccountNotification object:nil queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *notif) {
+        MXStrongifyAndReturnIfNil(self);
+#warning What was this doing? This should probably happen elsewhere
+        // self.onboardingCoordinatorBridgePresenter = nil;
         
-        [[NSNotificationCenter defaultCenter] removeObserver:authViewControllerObserver];
-        authViewControllerObserver = nil;
+        [[NSNotificationCenter defaultCenter] removeObserver:self.addAccountObserver];
+        self.addAccountObserver = nil;
     }];
     
-    authViewRemovedAccountObserver = [[NSNotificationCenter defaultCenter] addObserverForName:kMXKAccountManagerDidRemoveAccountNotification object:nil queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *notif) {
-        
+    self.removeAccountObserver = [[NSNotificationCenter defaultCenter] addObserverForName:kMXKAccountManagerDidRemoveAccountNotification object:nil queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *notif) {
+        MXStrongifyAndReturnIfNil(self);
         // The user has cleared data for their soft logged out account
-        _authViewController = nil;
+#warning What was this doing? This should probably happen elsewhere
+        // self.onboardingCoordinatorBridgePresenter = nil;
         
-        [[NSNotificationCenter defaultCenter] removeObserver:authViewRemovedAccountObserver];
-        authViewRemovedAccountObserver = nil;
+        [[NSNotificationCenter defaultCenter] removeObserver:self.removeAccountObserver];
+        self.removeAccountObserver = nil;
```

Comment:
**One big question**
I cannot figure out what `authViewControllerObserver` or `authViewRemovedAccountObserver` were actually doing. Using them to set the bridge presenter to `nil` breaks the key verification though as all the coordinators disappear 😕

Reply:
Good question I never pay attention to this observer. And there is no comments associated :/